torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -287,9 +287,97 @@ void removeNopPacking(std::shared_ptr<Graph>& graph) {
 
     n->removeAllInputs();
     it.destroyCurrent();
+
+    if (input->outputs()[0]->uses().size() == 0) {
+      input->destroy();
+    }
+  }
+}
+
+void fixDefaultRNNState(std::shared_ptr<Graph>& graph, Node * n, int input_index) {
+  auto initial_state = n->inputs()[input_index];
+  if (initial_state->node()->kind() == onnx::Constant ||
+      (initial_state->node()->kind() == onnx::Slice &&
+       initial_state->node()->inputs()[0]->node()->kind() == onnx::Constant)) {
```

Comment:
When is the default state a slice of a constant? A comment would be helpful

Reply:
++ will add comment